test/functional/system/transactions/expire_transactions.js

```diff
@@ -0,0 +1,173 @@
+		library = lib;
+		queries = new queriesHelper(lib, lib.db);
+	});
+
+	beforeEach('take mem accounts snapshot', () => {
+		return queries.getAccounts().then(accounts => {
+			memAccountsSnapshotBeforeUndo = _.cloneDeep(accounts);
+		});
+	});
+
+	it('should expire transaction from unconfirmed transaction list', done => {
+		const transaction = liskElements.transaction.transfer({
+			recipientId: randomUtil.account().address,
+			amount: randomUtil.number(100000000, 1000000000),
+			passphrase: accountsFixtures.genesis.passphrase,
+		});
+
+		localCommon.addTransactionToUnconfirmedQueue(library, transaction, () => {
+			// Wait for transaction to get expired and
+			// undo the unconfirmed transaction, so that mem accounts
+			// balance is updated
+			setTimeout(done, 10000);
+		});
+	});
+
+	it('should expire transaction from multi-signature transaction list', done => {
+		const transaction = liskElements.transaction.registerMultisignature({
+			passphrase: multiSig.account.passphrase,
+			keysgroup: multiSig.keysgroup,
+			lifetime: multiSig.lifetime,
+			minimum: multiSig.minimum,
+		});
+
+		// Get sender address
+		const address = getSenderAddress(transaction);
+
+		// Get multi-signature account created as part of
+		// transaction which is irreversible from mem account
+		const multiSigAccount = accountsFixtures.dbAccount({
+			address,
+			balance: new Bignum(transaction.amount).toString(),
+		});
+
+		// Set public key if not present
+		if (!multiSigAccount.publicKey) {
+			multiSigAccount.publicKey = Buffer.from(
+				transaction.senderPublicKey,
+				'hex'
+			);
+		}
+
+		// update mem account snapshot for comparision
+		memAccountsSnapshotBeforeUndo.push(multiSigAccount);
+
+		localCommon.addTransaction(library, transaction, () => {
+			// Wait for transaction to get expired and
+			// undo the unconfirmed transaction, so that mem accounts
+			// balance is updated
+			setTimeout(done, 1000);
+		});
+	});
+
+	describe('unconfirmed transaction', () => {
+		it('should be applied on mem accounts before transaction is expired', done => {
+			// set unconfirmed transaction timeout to 10 seconds
+			// to ensure the transaction is not expired for the
+			// next 10 seconds
+			setunconfirmedTransactionTimeOut(10000);
+
+			const amount = randomUtil.number(100000000, 1000000000);
+			const transaction = liskElements.transaction.transfer({
+				recipientId: randomUtil.account().address,
+				amount,
+				passphrase: accountsFixtures.genesis.passphrase,
+			});
+
+			const address = getSenderAddress(transaction);
+
+			localCommon.addTransactionToUnconfirmedQueue(library, transaction, () => {
+				queries
+					.getAccounts()
+					.then(memAccountsSnapshotAfterUndo => {
+						const beforeAccountState = memAccountsSnapshotBeforeUndo.find(
+							account => account.address === address
+						);
+						const afterAccountState = memAccountsSnapshotAfterUndo.find(
+							account => account.address === address
+						);
+						expect(
```

Comment:
same problem with expectations related to bignum

Reply:
fixed.